src/backend/x11/window.rs

```
@@ -0,0 +1,259 @@
+            .event_mask(
+                EventMask::EXPOSURE // Be told when the window is exposed
+            | EventMask::STRUCTURE_NOTIFY
+            | EventMask::KEY_PRESS // Key press and release
+            | EventMask::KEY_RELEASE
+            | EventMask::BUTTON_PRESS // Mouse button press and release
+            | EventMask::BUTTON_RELEASE
+            | EventMask::POINTER_MOTION // Mouse movement
+            | EventMask::RESIZE_REDIRECT // Handling resizes
+            | EventMask::NO_EVENT,
+            )
+            // Border pixel and color map need to be set if our depth may differ from the root depth.
+            .border_pixel(0)
+            .colormap(colormap);
+
+        let cookie = xcb.create_window(
+            depth.depth,
+            window,
+            screen.root,
+            0,
+            0,
+            properties.width,
+            properties.height,
+            0,
+            WindowClass::INPUT_OUTPUT,
+            visual_id,
+            &window_aux,
+        )?;
+
+        // Send requests to change window properties while we wait for the window creation request to complete.
+        let mut window = WindowInner {
+            connection: connection.clone(),
+            inner: window,
+            root: screen.root,
+            atoms,
+            size: Mutex::new((properties.width, properties.height).into()),
+            depth,
+            gc: 0,
+        };
+
+        let gc = xcb.generate_id()?;
+        xcb.create_gc(gc, window.inner, &CreateGCAux::new())?;
+        window.gc = gc;
+
+        // Enable WM_DELETE_WINDOW so our client is not disconnected upon our toplevel window being destroyed.
+        xcb.change_property32(
+            PropMode::REPLACE,
+            window.inner,
+            atoms.WM_PROTOCOLS,
+            AtomEnum::ATOM,
+            &[atoms.WM_DELETE_WINDOW],
+        )?;
+
+        // Block until window creation is complete.
+        cookie.check()?;
```

...Why? I know that everyone does this, but I never understood wait. Sure, this is not really expensive, but it is still one more round-trip to the X11 server.

And if this is about the error checking: Hm. How likely is it that creating the window fails? And the error is not silently lost anyway, it would then show up on the event loop instead...